src/js/worklets/onboarding_carousel.js

```diff
@@ -0,0 +1,60 @@
+import { useDerivedValue, withTiming, Easing } from 'react-native-reanimated';
+
+const slideAnimationDuration = 300;
+
+const easeOut = {
+  duration: slideAnimationDuration,
+  easing: Easing.bezier(0, 0, 0.58, 1),
+}
+
+// Derived Values
+export function dynamicProgressBarWidth (staticProgressBarWidth, progress) {
+  return useDerivedValue(
+    function () {
+      'worklet'
+      var progressValue = progress.value;
+      // Without this check, ios debug build is crashing
+      progressValue = isNaN(progressValue) ? 0 : progressValue;
+
+      return staticProgressBarWidth * progressValue / 100;
+    }
+  );
+}
+
+export function carouselLeftPosition (windowWidth, progress) {
+  return useDerivedValue(
+    function () {
+      'worklet'
+      var progressValue = progress.value;
+      switch (true) {
+      case (progressValue < 25):
+	return 0;
+	break;
+      case (progressValue == 25):
+	return withTiming(-windowWidth, easeOut);
+	break;
+      case (progressValue < 50):
+	return -windowWidth;
+	break;
+      case (progressValue == 50):
+	return withTiming(-2 * windowWidth, easeOut);
+	break;
+      case (progressValue < 75):
+	return -2 * windowWidth;
+	break;	
+      case (progressValue == 75):
+	return withTiming(-3 * windowWidth, easeOut);
+	break;
+      case (progressValue < 100):
+	return -3 * windowWidth;
+	break;	
+      case (progressValue == 100):
+	return withTiming(-4 * windowWidth, easeOut);
+	break;
+      default:
+	return 0;
+	break;
```

Comment:
Maybe improve indentation

```
switch(expression) {
  case x:
    // code block
    break;
  case y:
    // code block
    break;
  default:
    // code block
}
```

Reply:
Also `break` statements are not necessary as you are `return`ing right before `break`s

Reply:
> Maybe improve indentation

Can we do code formatting also for js files? cc @yqrashawn